pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -499,6 +499,19 @@ type Watermark struct {
 	// +kubebuilder:default="0s"
 	// +optional
 	MaxDelay *metav1.Duration `json:"maxDelay,omitempty" protobuf:"bytes,2,opt,name=maxDelay"`
+	// IdleSource defines the idle watermark properties, it could be configured in case source is idling.
+	// +kubebuilder:default={"maxWait": "-1s", "minIncrement": "-1s"}
+	// +optional
+	IdleSource *IdleSource `json:"idleSource,omitempty" protobuf:"bytes,3,opt,name=idleSource"`
+}
+
+type IdleSource struct {
+	// MaxWait is the wait duration before progressing the watermark in idle case.
+	// +kubebuilder:default="-1s"
+	MaxWait *metav1.Duration `json:"maxWait,omitempty" protobuf:"bytes,1,opt,name=maxWait"`
```

Comment:
protobuf index is not right.

Reply:
ohh, missed to update it while moving field arounds